src/operations/command.ts

```diff
@@ -73,12 +73,14 @@ export abstract class CommandOperation<
         : new MongoDBNamespace('admin', '$cmd');
     }
 
-    const propertyProvider = this.hasAspect(Aspect.NO_INHERIT_OPTIONS) ? undefined : parent;
-    this.readPreference = this.hasAspect(Aspect.WRITE_OPERATION)
-      ? ReadPreference.primary
-      : ReadPreference.resolve(propertyProvider, this.options);
-    this.readConcern = resolveReadConcern(propertyProvider, this.options);
-    this.writeConcern = resolveWriteConcern(propertyProvider, this.options);
+    const readPref = ReadPreference.fromOptions(options);
```

Comment:
I don't think so, since in `resolveOptions` when `ReadPreference.fromOptions` is undefined, we default to the parent read pref. If `fromOptions` always defaults to `primary`, we would never use the parent read pref.